src/mlir/consumer.h

```diff
@@ -0,0 +1,149 @@
+// Copyright Microsoft and Project Verona Contributors.
+// SPDX-License-Identifier: MIT
+
+#pragma once
+
+#include "generator.h"
+#include "mlir/IR/Builders.h"
+#include "mlir/IR/BuiltinOps.h"
+#include "mlir/IR/BuiltinTypes.h"
+#include "mlir/IR/MLIRContext.h"
+#include "parser/ast.h"
+
+#include "llvm/ADT/SmallVector.h"
+#include "llvm/ADT/StringRef.h"
+#include "llvm/Support/Error.h"
+
+#include <string>
+
+namespace mlir::verona
+{
+  /**
+   * Field offset maps between field names, types and their relative position.
+   *
+   * FIXME: This will need to be better encapsulated, but we'll expand it when
+   * we start implementing classes fully, and dynamic dispatch.
+   */
+  struct FieldOffset
+  {
+    llvm::SmallVector<llvm::StringRef, 1> fields;
+    llvm::SmallVector<Type, 1> types;
+  };
+
+  /**
+   * AST Consumer
```

Comment:
This needs a bit more explanation.  I presume this is the thing that produces MLIR from an AST, but AST consumer sounds like a generic interface for anything that can consume AST nodes.